rendermodules/point_match_optimization/point_match_optimization.py

```diff
@@ -0,0 +1,313 @@
+    tilepair_base_url += '&matchOwner=%s'%(render.DEFAULT_KWARGS['owner'])
+    tilepair_base_url += '&matchCollection=%s'%(collection_name)
+    tilepair_base_url += '&pId=%s'%(tile1)
+    tilepair_base_url += '&pGroupId=%s'%(pGroupId)
+    tilepair_base_url += '&qId=%s'%(tile2)
+    tilepair_base_url += '&qGroupId=%s'%(qGroupId)
+
+    return_struct = {}
+    return_struct['options'] = options
+    return_struct['keys'] = option_keys
+    return_struct['zipped'] = zip(option_keys, options)
+    return_struct['match_img_filename'] = match_img_filename
+    return_struct['collection_name'] = collection_name
+    return_struct['ptmatch_count'] = ptmatch_count
+    return_struct['tilepair_url'] = tilepair_base_url
+    return_struct['outdir'] = outdir
+    return return_struct
+
+
+class PointMatchOptimizationModule(RenderModule):
+    def __init__(self, schema_type=None, *args, **kwargs):
+        if schema_type is None:
+            schema_type = PointMatchOptimizationParameters
+        super(PointMatchOptimizationModule, self).__init__(
+            schema_type=schema_type, *args, **kwargs)
+
+    def run(self):
+        # create a stack with the two tiles
+        ts1 = renderapi.tilespec.get_tile_spec(self.args['stack'], self.args['tileId1'], render=self.render)
+        ts2 = renderapi.tilespec.get_tile_spec(self.args['stack'], self.args['tileId2'], render=self.render)
+        ts = []
+        ts.append(ts1)
+        ts.append(ts2)
+
+        pGroupId = ts1.layout.sectionId
+        qGroupId = ts2.layout.sectionId
+
+        # create the tile stack
+        if self.args['tile_stack'] is None:
+            tile_stack = "pmopt_{}_t{}".format(self.args['stack'],
+                                                time.strftime("%m%d%y_%H%M%S"))
+            self.args['tile_stack'] = tile_stack
+
+        renderapi.stack.create_stack(self.args['tile_stack'], render=self.render)
+        renderapi.client.import_tilespecs(self.args['tile_stack'], ts, render=self.render)
+        renderapi.stack.set_stack_state(self.args['tile_stack'], 'COMPLETE', render=self.render)
+
+        # the order of these two lines matter
+        keys, options = get_parameter_sets_and_strings(self.args['SIFT_options'])
+        mypartial = partial(compute_point_matches,
+                            self.render,
+                            self.args['tile_stack'],
+                            self.args['tileId1'],
+                            self.args['tileId2'],
+                            pGroupId,
+                            qGroupId,
+                            self.args['outputDirectory'],
+                            self.args['url_options'],
+                            keys)
+
+        with renderapi.client.WithPool(self.args['pool_size']) as pool:
+            return_struct = pool.map(mypartial, options)
+
+        #print(os.path.dirname(__file__))
+        m = render_from_template(os.path.dirname(__file__), 'optimization.html', return_struct=return_struct)
```

Comment:
This html should be included in the setup.py for distribution, I think in package_data?

Reply:
good catch russel.. yes this will fail when we deploy it on the cluster given the way that our deployment process works now.

Reply:
I think this is addressed now